### PR TITLE
Fixing integration tests

### PR DESCRIPTION
**Pull request by AdamSaleh**
So far:

[x] fixed koji mock in integration test
[x] removed F31
[x] fixed the ipslon mock based on @cverna s advice
[x] improved some of the exception handling
[x] only running integration tests for f33 for now

### CODE REVIEW

**Comment by pypingou** [comment]
> I don't think we install from pip anywhere

We don't but testing with pip allows to find out issues with new version of the dependencies before they make it to RPM and the stable Fedora releases.

**Comment by cverna** [comment]
> So far:
> 
> [x] fixed koji mock in integration test
> [x] removed F31
> 
> To do:
> 
> [ ] fix the ipslon mock, last 3 test consistently fail on 404 response from the ipsilon mock, logs suggest problem with badly initialized nonce?

for ipsilon I think you need to change the version of python in the Dockerfile. We manually copy a file and the path contains the python version see https://github.com/fedora-infra/bodhi/commit/43467b47f2c4e35ac06e23c2e70c18ce030ef19b

> 
> Possibly to do:
> 
> [ ] Disable more targets? i.e. I don't think we install from pip anywhere, and we might not even use F32? Need to doublecheck, but leaving just current stable (or current deployment target + current stable) should provide us with enough coverage.

File: devel/ci/integration/ipsilon/Dockerfile
Comment (cverna) [inline devel/ci/integration/ipsilon/Dockerfile]: we should probably tag this so that we are not surprised when the python version changes. I think `FROM fedora:33` would be better.

**Comment by cverna** [comment]
You will need to sign your commit for the DCO to pass :)

**Comment by AdamSaleh** [comment]
Plan is to have the two targets skipped and then tackle those after this is merged.

* https://github.com/fedora-infra/bodhi/issues/4158
* https://github.com/fedora-infra/bodhi/issues/4163

File: devel/ci/bodhi-ci
Comment (cverna) [inline devel/ci/bodhi-ci:1205]: integration tests are not blocking the merge, so we could in theory just leave them fail :(

File: devel/ci/integration/greenwave/start.sh
Comment (cverna) [inline devel/ci/integration/greenwave/start.sh]: hum that seems wrong, we should try to understand why it is missing

File: devel/ci/integration/tests/test_bodhi_cli.py
Comment (cverna) [inline devel/ci/integration/tests/test_bodhi_cli.py:485]: do we use 2 different binary for the koji cli ?

**Comment by cverna** [comment]
A few comments but other looks good

File: devel/ci/integration/tests/test_bodhi_cli.py
Comment (AdamSaleh) [reply devel/ci/integration/tests/test_bodhi_cli.py:485]: It seems that was f33/rawhide split? Or f32/f33? Can double-check.

File: devel/ci/integration/greenwave/start.sh
Comment (AdamSaleh) [reply devel/ci/integration/greenwave/start.sh]: I can try to pin the version of the image.

File: devel/ci/bodhi-ci
Comment (AdamSaleh) [reply devel/ci/bodhi-ci:1205]: No. In my experience, if they are failing and not blocking the merge, these are bad tests. 

Either they are important and should be passing and blocking. Or they are not important and shouldn't be there.

Comment (cverna) [reply devel/ci/bodhi-ci:1205]: Agree, I am fine going that way but we should consider the future of these tests, are we going to invest in them or not.

File: devel/ci/integration/tests/test_bodhi_cli.py
Comment (mattiaverga) [reply devel/ci/integration/tests/test_bodhi_cli.py:485]: It's because of https://pagure.io/koji/issue/912
Koji it's installed from RPM package, but it isn't found because it doesn't provide egg info. So Bodhi will install Koji from Pypi and the Koji-Pypi executable is used.

Comment (AdamSaleh) [reply devel/ci/integration/tests/test_bodhi_cli.py:485]: So, we keep the workaround? :-)

File: devel/ci/integration/greenwave/start.sh
Comment (AdamSaleh) [reply devel/ci/integration/greenwave/start.sh]: The image *is* pinned and after I removed it, I noticed integration tests still work. Reverted and will investigate in another issue.

File: devel/ci/bodhi-ci
Comment (AdamSaleh) [reply devel/ci/bodhi-ci:1205]: I am leaning in the way of "lets fix pip, keep just pip, rawhide and current stable and agressively(?) upgrade once new version is stable"

File: devel/ci/integration/tests/test_bodhi_cli.py
Comment (cverna) [reply devel/ci/integration/tests/test_bodhi_cli.py:485]: Thanks @mattiaverga for finding that issue, I knew I saw it in the past :+1: 

Yeah let's keep the workaround, maybe we can add the koji issue as comment to explain why we did that ?

Comment (AdamSaleh) [reply devel/ci/integration/tests/test_bodhi_cli.py:485]: Ok, will add Mattia's comment.

**Review by cverna — APPROVED** [review]

**Pull request merged**
